### PR TITLE
New 'limited admin' role

### PR DESCRIPTION
**Pull request by jits**
Currently only granted read access to the costs reports listing and details pages.

### CODE REVIEW

File: platform-hub-api/app/models/ability.rb [hunk 1]
```diff
@@ -6,10 +6,18 @@ def initialize(user)
     # IMPORTANT: to understand the caveats when using blocks to specify
     # abilities, see: https://github.com/CanCanCommunity/cancancan/wiki/Defining-Abilities-with-Blocks
 
-    # This will take precedence over everything below!
+
+    # Hub admin role
+    # Note: this will take precedence over everything below!
```
Comment (KashifSaadat) [inline platform-hub-api/app/models/ability.rb]: Is this comment still relevant, as you can only have one of the roles active now?

File: platform-hub-api/spec/rails_helper.rb [hunk 2]
```diff
@@ -66,13 +66,11 @@
...
   # Support and shared stuff
-  config.include JsonResponseHelper, type: :request
   config.include JsonResponseHelper, type: :controller
   config.include UserActivationHelpers, type: :controller
-  config.include_context 'authentication helpers', type: :request
   config.include_context 'authentication helpers', type: :controller
-  config.include_context 'admin authorization helpers', type: :request
-  config.include_context 'admin authorization helpers', type: :controller
+  # config.include_context 'hub admin authorization helpers', type: :controller
+  # config.include_context 'hub limited admin authorization helpers', type: :controller
```
Comment (KashifSaadat) [inline platform-hub-api/spec/rails_helper.rb]: These still needed?

File: platform-hub-web/src/app/app.routes.js [hunk 3]
```diff
@@ -771,15 +771,15 @@ export const appRoutes = function ($stateProvider, $urlRouterProvider, $location
...
+          rolesPermitted: ['admin', 'limited_admin']
         }
       })
       .state('costs-reports.new', {
         url: '/new',
         component: CostsReportsForm,
         data: {
           authenticate: true,
-          rolePermitted: 'admin'
+          rolesPermitted: ['admin']
```
Comment (KashifSaadat) [inline platform-hub-web/src/app/app.routes.js]: Would we also want to give `limited_admin` people the ability to create new cost reports?

File: platform-hub-api/app/models/ability.rb (hunk 1 above)
Comment (jits) [reply platform-hub-api/app/models/ability.rb]: It's still relevant, but just in the context of how `cancancan` uses this file – i.e. the _ordering_ of these abilities/rules are really important.

File: platform-hub-api/spec/rails_helper.rb (hunk 2 above)
Comment (jits) [reply platform-hub-api/spec/rails_helper.rb]: ah good catch, I need to remove

File: platform-hub-web/src/app/app.routes.js (hunk 3 above)
Comment (jits) [reply platform-hub-web/src/app/app.routes.js]: I don't believe we'll be doing that at the moment – I think it's purely for consumption at this stage (though it's easy to add access for it later, if needed).

**Pull request merged**